### PR TITLE
[CiApp] Include .NET 6.0 in dd-trace tool

### PR DESCRIPTION
**Pull request by tonyredondo**
@DataDog/apm-dotnet

### CODE REVIEW

**Review by pierotibou — APPROVED** [review]

File: tracer/src/Datadog.Trace.Tools.Runner/Datadog.Trace.Tools.Runner.Standalone.csproj [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <Title>Datadog APM Auto-instrumentation Runner</Title>
     <Copyright>Copyright 2020 Datadog, Inc.</Copyright>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
+    <TargetFrameworks>net6.0</TargetFrameworks>
```
Comment (andrewlock) [inline tracer/src/Datadog.Trace.Tools.Runner/Datadog.Trace.Tools.Runner.Standalone.csproj:8]: Any reason we didn't update this to `net5.0` before? 🤔 

AFAIK `net5.0` and `net6.0` use a different single-file behaviour to `netcoreapp3.1`, namely that native files aren't included in the bundle by default (i.e. it's not actually a single file). Should we set `IncludeNativeLibrariesForSelfExtract`? Or even `IncludeAllContentForSelfExtract`?

Also, should we [enable compression for the single file](https://docs.microsoft.com/en-us/dotnet/core/deploying/single-file#compress-assemblies-in-single-file-app)? that has a startup cost, but will reduce file size.

**Review by andrewlock — APPROVED** [review]
I think we should double check the output of the standalone runner, as there are differences in the files that will be generated in netcoreapp3.1 vs net6.0

File: tracer/src/Datadog.Trace.Tools.Runner/Datadog.Trace.Tools.Runner.Standalone.csproj (hunk 1 above)
Comment (tonyredondo) [reply tracer/src/Datadog.Trace.Tools.Runner/Datadog.Trace.Tools.Runner.Standalone.csproj:8]: hmmm 🤔 that's a good one... why we didn't change it for .NET 5.0???.... I don't recall any reason...

I'll check both properties, thank you. 

The goal for the PR was adding .NET 6.0 build in the tool.
image

But let's solve how we are going to do it for the standalone version as well...

Comment (tonyredondo) [reply tracer/src/Datadog.Trace.Tools.Runner/Datadog.Trace.Tools.Runner.Standalone.csproj]: I reverted the target framework to `netcoreapp3.1`  for the standalone version, I'll do another PR for this. I prefer to merge this and include .NET 6 to the dd-trace tool first.

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2206 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/7488004eef7732be5a53354b0c8e5c798ce9ca2d) will **not change** line coverage
:heavy_check_mark: Merging #2206 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/7488004eef7732be5a53354b0c8e5c798ce9ca2d) will **not change** branch coverage
:heavy_check_mark: Merging #2206 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/7488004eef7732be5a53354b0c8e5c798ce9ca2d) will **not change** complexity


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/7488004eef7732be5a53354b0c8e5c798ce9ca2d) | #2206       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `9312` / `12779` | `9306` / `12779` |          |
| Lines %   | `73%`      | `73%`      |  `0%` :heavy_check_mark:  |
| Branches  | `4233` / `6236` | `4230` / `6236` |          |
| Branches %| `68%`      | `68%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `7432`      | `7432`        |  `0`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84479&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84479/artifacts?artifactName=Code%20Coverage%20Report_84479&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2206](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84504&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84504/artifacts?artifactName=Code%20Coverage%20Report_84504&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/7488004eef7732be5a53354b0c8e5c798ce9ca2d) | #2206       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `73%`      | `73%`       |  `0%` :heavy_check_mark:  |
| Branches %| `68%`      | `68%`       |  `0%` :heavy_check_mark:  |
| Complexity| `7432`      | `7432`       |  `0` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Couchbase.IIOServiceExecuteIntegrationBis](https://github.com/DataDog/dd-trace-dotnet/tree/34a72d8c2f162fded215c6be62eeb2451fed366b/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Couchbase/IIOServiceExecuteIntegrationBis.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.MsTestV2.TestMethodRunnerExecuteIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/34a72d8c2f162fded215c6be62eeb2451fed366b/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteIntegration.cs) | `-33%` :no_entry: | `-19%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.Agent.CIAgentWriter](https://github.com/DataDog/dd-trace-dotnet/tree/34a72d8c2f162fded215c6be62eeb2451fed366b/tracer/src/Datadog.Trace/Ci/Agent/CIAgentWriter.cs) | `-10%` :no_entry: | `-21%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.NUnit.NUnitCompositeWorkItemSkipChildrenIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/34a72d8c2f162fded215c6be62eeb2451fed366b/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs) | `-8%` :no_entry: | `-8%` :no_entry:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84479&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84479/artifacts?artifactName=Code%20Coverage%20Report_84479&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2206](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84504&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84504/artifacts?artifactName=Code%20Coverage%20Report_84504&api-version=6.0&%24format=zip)

**Review by lucaspimentel — APPROVED** [review]

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2206 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/5344edb1cb50ba24cb062935c928c3a4d4bb5b99) will **not change** line coverage
:heavy_check_mark: Merging #2206 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/5344edb1cb50ba24cb062935c928c3a4d4bb5b99) will **not change** branch coverage
:heavy_check_mark: Merging #2206 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/5344edb1cb50ba24cb062935c928c3a4d4bb5b99) will will **decrease** complexity by `25`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/5344edb1cb50ba24cb062935c928c3a4d4bb5b99) | #2206       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `9222` / `12597` | `9185` / `12556` |          |
| Lines %   | `73%`      | `73%`      |  `0%` :heavy_check_mark:  |
| Branches  | `4205` / `6160` | `4188` / `6132` |          |
| Branches %| `68%`      | `68%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `7357`      | `7332`        |  `-25`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84611&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84611/artifacts?artifactName=Code%20Coverage%20Report_84611&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2206](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84647&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84647/artifacts?artifactName=Code%20Coverage%20Report_84647&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/5344edb1cb50ba24cb062935c928c3a4d4bb5b99) | #2206       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `73%`      | `73%`       |  `0%` :heavy_check_mark:  |
| Branches %| `68%`      | `68%`       |  `0%` :heavy_check_mark:  |
| Complexity| `7357`      | `7332`       |  `-25` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Couchbase.IIOServiceExecuteIntegrationBis](https://github.com/DataDog/dd-trace-dotnet/tree/6acc3af55fcfadaaad5d5b6bb59c0d76a3a553bc/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Couchbase/IIOServiceExecuteIntegrationBis.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.MsTestV2.TestMethodRunnerExecuteIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/6acc3af55fcfadaaad5d5b6bb59c0d76a3a553bc/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteIntegration.cs) | `-33%` :no_entry: | `-19%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.Agent.CIAgentWriter](https://github.com/DataDog/dd-trace-dotnet/tree/6acc3af55fcfadaaad5d5b6bb59c0d76a3a553bc/tracer/src/Datadog.Trace/Ci/Agent/CIAgentWriter.cs) | `-10%` :no_entry: | `-21%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.NUnit.NUnitCompositeWorkItemSkipChildrenIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/6acc3af55fcfadaaad5d5b6bb59c0d76a3a553bc/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs) | `-8%` :no_entry: | `-8%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.SpanContextPropagator](https://github.com/DataDog/dd-trace-dotnet/tree/6acc3af55fcfadaaad5d5b6bb59c0d76a3a553bc/tracer/src/Datadog.Trace/SpanContextPropagator.cs) | `10%` :heavy_check_mark: | `12%` :heavy_check_mark:   | `-25` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84611&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84611/artifacts?artifactName=Code%20Coverage%20Report_84611&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2206](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84647&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84647/artifacts?artifactName=Code%20Coverage%20Report_84647&api-version=6.0&%24format=zip)

**Pull request merged**